oci/layout/oci_archive_dest.go

```diff
@@ -0,0 +1,126 @@
+package layout
```

Comment:
`oci/layout.NewReference` should be all that is necessary AFAICT.

Then, perhaps, some utility subpackage to minimize the amount of copy&pasted parsing/formatting code shared between `oci/layout` and `oci/archive`, `ValidatePolicyConfigurationScope` and the like — depending on the details of how the two end up.